### PR TITLE
Removed AdminParent controllers from search in BackOffice

### LINKED ISSUE #26565: While searching a page in the BO, I have a not found page as a result
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

While searching a page in the BO (orders for example), I have 2 results.

One of them is a link to the adminparent controller: the first tab in the menu which is not clickable and leads to a not found page when we click on it.
That's a regression: in 1.7.8, no results were found.

Capture d’écran 2021-11-10 à 10 15 38

Capture d’écran 2021-11-10 à 10 18 59

### Expected behavior

The adminparent controller shouldn't be listed as a result.

### Steps to reproduce

1. Go to the BO
2. Search for a migrated page (eg: orders)
3. Click on the two results
4. See that one is a not found page

### PrestaShop version(s) where the bug happened

V8 (develop branch)

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Removed AdminParent controllers from search in BackOffice
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #26565
| How to test?      | Cf. #26565

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27012)

### CODE REVIEW

File: tests/Integration/Controllers/AdminSearchControllerCoreTest.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            [
+                'orders',
+                [
+                    'query' => 'orders',
+                    'show_toolbar' => true,
+                    'nb_results' => 2,
+                    'features' => [
+                        'Orders' => [
+                            [
+                                'link' => 'http://localhost/admin-dev/index.php?controller=AdminOrders&token=dc20fd6b3135f9fc1a592a38a3aff1c9',
```
Comment (sowbiba) [inline tests/Integration/Controllers/AdminSearchControllerCoreTest.php]: why adding a token in the result if you clean it in the assert ?

**Review by sowbiba — DISMISSED** [review]
Just a question, not blocking

File: tests/Integration/Controllers/AdminSearchControllerCoreTest.php (hunk 1 above)
Comment (matthieu-rolland) [reply tests/Integration/Controllers/AdminSearchControllerCoreTest.php]: Indeed it seems superfluous, I still approve the PR since it's not blocking

**Review by matthieu-rolland — DISMISSED** [review]

File: tests/Integration/Controllers/AdminSearchControllerCoreTest.php (hunk 1 above)
Comment (Progi1984) [reply tests/Integration/Controllers/AdminSearchControllerCoreTest.php]: Done

**Review by matthieu-rolland — DISMISSED** [review]
LGTM 👍

**Review by atomiix — DISMISSED** [review]

**Comment by HanaRebaiQA** [comment]
Hello @Progi1984 

I have checked this PR. The issue is not fixed.
You can check here with PR (eg.serach with orders / products) : 
image

image

https://watch.screencastify.com/v/Zi5SsoKKmcAhL9txKSGV
PS1778 : 
https://watch.screencastify.com/v/BTJGhhZECuMFwWB8y7kn

Please check and feedback!

Thanks!

**Comment by Progi1984** [comment]
@HanaRebaiQA Ok, it's out of scope of the PR for `AdminSupplyOrders` : it shouldn't be in the core since 1.7.2.0 RC1. I have created an another PR #27242 for removing remaining code about `AdminSupplyOrders`.

**Comment by HanaRebaiQA** [comment]
> AdminSupplyOrders

@Progi1984 : The problem is not related only to `AdminSupplyOrders`, but even with `AdminModulesCatalog` which are well functional on PS1778. Please see the screenrecord above.

**Comment by Progi1984** [comment]
@HanaRebaiQA It's normal. In v`8.0.0`, there is not anymore `AdminModulesCatalog` which is in the `MBO` module managed by the company PrestaShopCorp (FYI @intraordinaire).

**Comment by HanaRebaiQA** [comment]
Hello @Progi1984 

Thanks for your feedback! Actually, Yes i found a lot of not found pages related to AdminCatalog, AdminSUpplyOrders, AdminModulesCatalog, etc
But for `AdminParent` controllers still not fixed when searching for `Theme`. There is two result of `Theme & Logo`, which the first it's functional and the second one it's related to the controller `AdminThemesParent`.
image
image

https://watch.screencastify.com/v/Gljlk83YI7BwHg6DoxEh

Please check and feedback!

Thanks!

File: controllers/admin/AdminSearchController.php [hunk 2]
```diff
@@ -250,22 +250,45 @@ public function searchFeatures()
...
+            _DB_PREFIX_,
+            (int) $this->context->employee->id_lang
         );
+        $result = Db::getInstance()->executeS($sql);
+        $mainControllers = Dispatcher::getControllers([
+            _PS_ADMIN_DIR_ . '/tabs/',
+            _PS_ADMIN_CONTROLLER_DIR_,
+            _PS_OVERRIDE_DIR_ . 'controllers/admin/',
+        ]);
+        //dump($mainControllers);
```
Comment (atomiix) [inline controllers/admin/AdminSearchController.php]:
```suggestion
```

Comment (Progi1984) [reply controllers/admin/AdminSearchController.php]: Fixed. Thanks.

**Review by PierreRambaud — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by HanaRebaiQA** [comment]
Hello @Progi1984 

Thanks for the fix. 
I checked again the PR and there is no `AdminParent` Controller. 
Only `AdminModulesCatalog` still present.

So , QA :heavy_check_mark: 

I have verified with many langages : For example
image

Thanks!

**Comment by Progi1984** [comment]
Thanks @HanaRebaiQA

**Pull request merged**